src/test/java/io/jenkins/plugins/report/jtreg/main/comparator/PrintTableTest.java

```diff
@@ -0,0 +1,34 @@
+package io.jenkins.plugins.report.jtreg.main.comparator;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+public class PrintTableTest {
+    @Test
+    public void testTablePrinting() {
+        final ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+        final PrintStream originalStream = System.out;
+        System.setOut(new PrintStream(outStream));
+
+        String[][] table = {
+                {null, "first item", "second item", "third item"},
+                {"second row", "X", "this is a very long text in a center cell of a table", "X"},
+                {"third row", null, "", "X"},
+                {"fourth row", "X", "X", "X", "this is out of range and it won't be shown"}
+        };
+
+        PrintTable.print(table, 4, 4);
+
+        Assert.assertEquals("1) first item\n" +
+                "2) second item\n" +
+                "3) third item\n" +
+                "           | 1 | 2                                                    | 3 | \n" +
+                "second row | X | this is a very long text in a center cell of a table | X | \n" +
+                "third row  |   |                                                      | X | \n" +
+                "fourth row | X | X                                                    | X | \n", outStream.toString());
+        System.setOut(originalStream);
```

Comment:
Where this is highly  appreciated, it is very fragile.
If you would have class taking care bout Inpouts (that  print/printe methods), you would be able to simply swap it fro ByteArray implementation, instead of ffiddling with stdout/stderr

Please reset the streams back in @afterClass ; this is not negotiable.


Another better thing may be to check internl representatiion before printing. You surely hve it.
But that do not replace this test. It is another, and mandatory one, test before this one. This one is chekcing it is printed out correclty, not that it was processed correctly. The asserts aroundinternal structures before printing are the ones veryfing it was calcualted correctly.

Reply:
This test is mainly for testing if the PrintTable class prints the data correctly (prints the table with right spaces, etc). .I moved the stream setting and reseting to `@BeforeClass` and `@AfterClass`.